### PR TITLE
Improve per cache tutorial

### PR DESCRIPTION
**Pull request by karesti**
Add configuration by URI

### CODE REVIEW

**Review by oraNod — APPROVED** [review]
One minor thing about the Mac workaround but otherwise LGTM

File: remote-per-cache/src/main/java/org/infinispan/tutorial/simple/remote/percache/InfinispanRemotePerCache.java [hunk 1]
```diff
@@ -30,15 +31,21 @@ public static void main(String[] args) {
              .security().authentication()
                //Add user credentials.
                .username("admin")
-               .password("password")
-               .realm("default")
-               .saslMechanism("DIGEST-MD5");
+               .password("password");
+
+      // Docker 4 Mac work-around, don't change this in production
+      builder.clientIntelligence(ClientIntelligence.BASIC);
```
[the remark this answers is not in the document]
Comment (karesti) [reply remote-per-cache/src/main/java/org/infinispan/tutorial/simple/remote/percache/InfinispanRemotePerCache.java:37]: That's fine for me, if we keep it in a single place (see my other PR)

[the remark this answers is not in the document]
Comment (karesti) [reply remote-per-cache/src/main/java/org/infinispan/tutorial/simple/remote/percache/InfinispanRemotePerCache.java:37]: Well, the problem with the issue with docker 4 mac is that it does not fail. It just hangs without error so it's hard to debug. Unless we find a way to check the OS and it's docker 4 mac add it ... hm

[the remark this answers is not in the document]
Comment (oraNod) [reply remote-per-cache/src/main/java/org/infinispan/tutorial/simple/remote/percache/InfinispanRemotePerCache.java:37]: What if we change the comment to something like "Use BASIC client intelligence for a locally running server."?

I'm not sure we need to point out that it's not for production. And making the comment more generic, less Mac specific, might be good.

[the remark this answers is not in the document]
Comment (karesti) [reply remote-per-cache/src/main/java/org/infinispan/tutorial/simple/remote/percache/InfinispanRemotePerCache.java:37]: The thing is that this is docker 4 mac specific issue. you don't need this if you are using linux

[the remark this answers is not in the document]
Comment (karesti) [reply remote-per-cache/src/main/java/org/infinispan/tutorial/simple/remote/percache/InfinispanRemotePerCache.java:37]: maybe for windows as well but I did not test on windows

[the remark this answers is not in the document]
Comment (oraNod) [reply remote-per-cache/src/main/java/org/infinispan/tutorial/simple/remote/percache/InfinispanRemotePerCache.java:37]: ok. should we go ahead and merge this?

[the remark this answers is not in the document]
Comment (karesti) [reply remote-per-cache/src/main/java/org/infinispan/tutorial/simple/remote/percache/InfinispanRemotePerCache.java:37]: yes please

**Comment by oraNod** [comment]
thanks @karesti

**Pull request merged**